@uportal/esco-content-menu-lit/README.md

```diff
@@ -0,0 +1,643 @@
+>
+  <div slot="footer"></div>
+</esco-content-grid>
+```
+
+#### Theming
+
+This component supports [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables) for overriding some default values. So defining the following variables will override default values.
+
+```css
+:root {
+  --content-grid-flex-grid-justify: center; // How to justify all flex grid elements, default value is center
+  --content-grid-flex-grid-item-margin: 20px auto; // To set a margin on all flex items, default is `20px auto`
+}
+```
+
+### The action favorite
+
+The component `action-favorite` is really simple, it show a start button that permit to add or remove from favorites a portlet.
+
+#### Html
+
+```html
+<script
+  src="/resource-server/webjars/uportal__esco-content-menu/dist/esco-content-menu.min.js"
+  defer
+></script>
+
+<esco-action-favorite portlet-card-size="small" hide-action="false">
+</esco-action-favorite>
+```
+
+#### Properties
+
+- `messages`: type: `Array of Objects`, see [internationalization](#internationalization)
+- ~~`call-on-toggle-fav`: type: `Function`, default: `{}`, a callback function called after the click event on the button,~~ Deprecated, use `toggle` event instead.
+- `chan-id`: type: `String`, required: `true`, the portlet id to add or remove from user favorites,
+- ~~`favorite-api-url`: type: `String`, default: `/uPortal/api/layout`, the uri/url of the favorites api,~~ Deprecated.
+- ~~`user-info-api-url`: type: `String`, default: `/uPortal/api/v5-1/userinfo`, url/uri on which the api request is done to obtain user information and the jwt token,~~ Deprecated.
+- `fname`: type: `String`, required: `true`, the portlet fname that permit to identify the portlet into favorite's list, usefull for the callback function and apply a css class,
+- `is-favorite`: type: `Boolean`, default: `false`, provide the favorite state,
+- `back-ground-is-dark`: type: `Boolean`, default: `false`, permit to apply a style depending on background color, as the component is used as embeded,
+- ~~`debug`: type: `Boolean`, default: `false`, for the demo/debug mode to be able to run in a standalone way (disable favorites api call)~~ Deprecated
+
+#### Events
+
+##### Emitted
+
+- `toggle`, target : `component`, details: `fname,chanId,isFavorite`, emited when button is clicked.
+
+#### Theming
+
+This component supports [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables) for overriding some default values. So defining the following variables will override default values.
+
+```css
+:root {
+  --action-favorites-dark-background-icon-color: #010101; // To set a different color of the favorite svg icon when background is dark.
+  --action-favorites-normal-icon-color: red; // To set a different color of the favorite svg icon when background is normal.
+}
+```
+
+### The content favorites
+
+The component is functional only into the `content-menu`, it needs that a portlet list and favorite list is passed
```

Comment:
What is 'portlet list', ie which api does it come from?

Reply:
The `portlet list` is only an array passed to the component, so it's resolved from a parent component and using a service class that resolve from the current portlet api system.